spiped/dispatch.c

```diff
@@ -137,20 +166,38 @@ callback_gotconn(void * cookie, int s)
 	if ((sas = sock_addr_duplist(A->sas)) == NULL)
 		goto err1;
 
+	/* Create new conn_list_node. */
+	if ((node_new = malloc(sizeof(struct conn_list_node))) == NULL)
+		goto err2;
+	node_new->prev = NULL;
+	node_new->next = NULL;
+	node_new->A = A;
+
 	/* Create a new connection. */
-	if (proto_conn_create(s, sas, A->decr, A->nofps, A->requirefps,
-	    A->nokeepalive, A->K, A->timeo, callback_conndied, A) == NULL) {
+	if ((node_new->conn_cookie = proto_conn_create(s, sas, A->decr,
+	    A->nofps, A->requirefps, A->nokeepalive, A->K, A->timeo,
+	    callback_conndied, node_new)) == NULL) {
 		warnp("Failure setting up new connection");
-		goto err2;
+		goto err3;
 	}
 
+	/* Link node_new to the beginning of the conn_cookies list. */
+	if (A->conn_cookies != NULL)
+		A->conn_cookies->prev = node_new;
```

Comment:
Since you're initializing `node_new->next = NULL` earlier, you could put both of these lines under the `if`.

Also, I like spelling it `node_new->next->prev = node_new`.